dropwizard-auth/src/main/java/io/dropwizard/auth/Authorizer.java

```diff
@@ -34,4 +34,15 @@
     default boolean authorize(P principal, String role, @Nullable ContainerRequestContext requestContext) {
         return authorize(principal, role);
     }
+
+    /**
+     * Returns an {@link AuthorizationContext} object, to be used in {@link CachingAuthorizer} as cache key.
+     * @param principal      a {@link Principal} object, representing a user
+     * @param role           a user role
+     * @param requestContext a request context.
+     * @return {@link AuthorizationContext} object, to be used in {@link CachingAuthorizer}.
+     */
```

Comment:
Can do, but I was planning to backport it to 2.0 eventually, unless you'd rather not?

Reply:
@cyberdelia Given the change in the public API with adding this method, I would only add the change to Dropwizard 2.1.0.

Otherwise existing implementations with the same method and parameters but different return type might fail after a patch release.

See https://wiki.eclipse.org/Evolving_Java-based_APIs_2#Evolving_API_Interfaces for a rough guideline.

Reply:
Done!